src/renderer/services/plugin-manager.js

```diff
@@ -67,11 +68,23 @@ class PluginManager {
       profileId
     })
 
-    await this.loadComponents(pluginObject, plugin)
-    await this.loadRoutes(pluginObject, plugin)
-    await this.loadMenuItems(pluginObject, plugin, profileId)
-    await this.loadAvatars(pluginObject, plugin, profileId)
-    await this.loadWalletTabs(pluginObject, plugin, profileId)
+    const permissions = zipObject(plugin.config.permissions, fill(new Array(plugin.config.permissions.length), true))
```

Comment:
This feels like a long-winded way of building one object when the object already exists. Why not just 

```
if (plugin.config.permissions.includes('COMPONENTS')) {

}
```

If you're insistent on doing the zip stuff, do it when first loading the JSON so the "zipping" isn't done every time the plugin is enabled

Reply:
Using includes would iterate over the array on every check, that's why I'm building an object with the available permissions, to iterate over the array only once.

I though that the correct place to check permissions, was when enabling the plugin, @alexbarnsley, where would you do it?

Reply:
I agree that the correct place to check for permissions is when enabling. But I think the `zipObject` response needs to be stored on the plugin object itself so it doesn't get called every time the plugin is enabled. Possibly here: https://github.com/ArkEcosystem/desktop-wallet/blob/9b0401831143f13c5ada0499eac63106162a5d90/src/renderer/services/plugin-manager.js#L459

```
    const permissions = zipObject(config.permissions, fill(new Array(plugin.config.permissions.length), true))

    config = {
      ...,
      permissions,
      ...
    }
```

Reply:
I'm not sure about that, because it's more space-efficient using an Array to store that data, than storing an Object, which values are always `true`.

Using `zipObject` each time a user enables a plugin, something that it's not going to be very common, is not expensive enough to justify using that additional space.

The only use of that line is to avoid searching later, on each permission, if it can load data or not; as an alternative implementation, I could have used `_.difference` to obtain only the permissions that apply and then use the `loadX` method.